### ISSUE TITLE
media-configdrive.mount is failing in systemd v229

### ISSUE THREAD

**Issue by mischief** [coreos/bugs#1167]
in a test image built with systemd 229 running under QEMU, media-configdrive.mount fails, and did not fail or even execute with systemd 225.

```
$ ssh coretest
Last login: Wed Mar 16 01:39:16 2016
CoreOS developer (983.0.0+2016-03-15-1459)
Failed Units: 1
  media-configdrive.mount
core@coreos_production_qemu-983-0-0-2 ~ $ systemctl cat media-configdrive.mount
# /usr/lib64/systemd/system/media-configdrive.mount
[Unit]
Wants=user-configdrive.service
Before=user-configdrive.service
# Only mount config drive block devices automatically in virtual machines
# or any host that has it explicitly enabled and not explicitly disabled.
ConditionVirtualization=|vm
ConditionKernelCommandLine=|coreos.configdrive=1
ConditionKernelCommandLine=!coreos.configdrive=0

[Mount]
What=LABEL=config-2
Where=/media/configdrive
Options=ro
core@coreos_production_qemu-983-0-0-2 ~ $ journalctl -b -u media-configdrive.mount --no-pager
-- Logs begin at Wed 2016-03-16 01:39:07 UTC, end at Wed 2016-03-16 01:46:00 UTC. --
Mar 16 01:39:10 localhost systemd[1]: Mounting /media/configdrive...
Mar 16 01:39:10 localhost mount[554]: mount: can't find LABEL=config-2
Mar 16 01:39:10 localhost systemd[1]: media-configdrive.mount: Mount process exited, code=exited status=1
Mar 16 01:39:10 localhost systemd[1]: Failed to mount /media/configdrive.
Mar 16 01:39:10 localhost systemd[1]: media-configdrive.mount: Unit entered failed state.
core@coreos_production_qemu-983-0-0-2 ~ $ cat /proc/cmdline 
rootflags=rw mount.usrflags=ro BOOT_IMAGE=/coreos/vmlinuz-a mount.usr=PARTUUID=7130c94a-213a-4e5a-8e26-6cce9662f132 rootflags=rw mount.usrflags=ro consoleblank=0 root=LABEL=ROOT console=ttyS0,115200n8 console=tty0 coreos.first_boot=1 coreos.randomize_disk_guid=00000000-0000-0000-0000-000000000001 verity.usrhash=2b3e5ee787d6cd362f04599c226e7be8ab38d0cb79ad6f73a47cada7f0740573
core@coreos_production_qemu-983-0-0-2 ~ $ systemctl --version
systemd 229
-PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT -GNUTLS -ACL +XZ -LZ4 +SECCOMP +BLKID -ELFUTILS +KMOD -IDN
```

**Comment by mischief** [comment]
something has gone wrong with the conditions, so the unit starts anyway:

```
Mar 16 17:24:45 localhost systemd[1]: media-configdrive.mount: ConditionKernelCommandLine=!coreos.configdrive=0 succeeded.
Mar 16 17:24:45 localhost systemd[1]: media-configdrive.mount: ConditionKernelCommandLine=|coreos.configdrive=1 failed.
Mar 16 17:24:45 localhost systemd[1]: media-configdrive.mount: ConditionVirtualization=|vm succeeded.
Mar 16 17:24:45 localhost systemd[1]: media-configdrive.mount: Failed to check symlink /media/configdrive, ignoring: No such file or directory
Mar 16 17:24:45 localhost systemd[1]: media-configdrive.mount: About to execute: /bin/mount LABEL=config-2 /media/configdrive -o ro
Mar 16 17:24:45 localhost systemd[1]: media-configdrive.mount: Forked /bin/mount as 515
Mar 16 17:24:45 localhost systemd[1]: media-configdrive.mount: Changed dead -> mounting
Mar 16 17:24:45 localhost systemd[1]: Mounting /media/configdrive...
Mar 16 17:24:45 localhost systemd[515]: media-configdrive.mount: Executing: /bin/mount LABEL=config-2 /media/configdrive -o ro
```

**Comment by mischief** [comment]
i think `user-configdrive.path` is now triggering the mount unit due to new Requires/After properties. i think systemd/systemd@c7c89abb9edf9320246482bf4a8e0656199281ae introduced this behavior but i'm not sure.

```diff
--- good.path	2016-03-16 11:30:09.188892068 -0700
+++ bad.path	2016-03-16 11:30:36.029027205 -0700
@@ -5,24 +5,23 @@
 Result=success
 Id=user-configdrive.path
 Names=user-configdrive.path
-Requires=media.mount sysinit.target
+Requires=media-configdrive.mount media.mount sysinit.target
 RequiredBy=user-config.target
 Conflicts=shutdown.target
-Before=paths.target user-configdrive.service user-config.target shutdown.target
-After=-.mount media.mount sysinit.target
+Before=user-config.target shutdown.target user-configdrive.service paths.target
+After=media-configdrive.mount sysinit.target media.mount -.mount
 Triggers=user-configdrive.service
 RequiresMountsFor=/media/configdrive
 Description=Watch for a cloud-config at /media/configdrive
 LoadState=loaded
-ActiveState=active
-SubState=waiting
+ActiveState=inactive
+SubState=dead
 FragmentPath=/usr/lib64/systemd/system/user-configdrive.path
 UnitFileState=static
 UnitFilePreset=disabled
-InactiveExitTimestamp=Wed 2016-03-16 18:29:12 UTC
-InactiveExitTimestampMonotonic=5839812
-ActiveEnterTimestamp=Wed 2016-03-16 18:29:12 UTC
-ActiveEnterTimestampMonotonic=5839812
+StateChangeTimestampMonotonic=0
+InactiveExitTimestampMonotonic=0
+ActiveEnterTimestampMonotonic=0
 ActiveExitTimestampMonotonic=0
 InactiveEnterTimestampMonotonic=0
 CanStart=yes
@@ -36,15 +35,15 @@
 DefaultDependencies=yes
 OnFailureJobMode=replace
 IgnoreOnIsolate=no
-IgnoreOnSnapshot=no
 NeedDaemonReload=no
-JobTimeoutUSec=0
+JobTimeoutUSec=infinity
 JobTimeoutAction=none
-ConditionResult=yes
-AssertResult=yes
-ConditionTimestamp=Wed 2016-03-16 18:29:12 UTC
-ConditionTimestampMonotonic=5839774
-AssertTimestamp=Wed 2016-03-16 18:29:12 UTC
-AssertTimestampMonotonic=5839774
+ConditionResult=no
+AssertResult=no
+ConditionTimestampMonotonic=0
+AssertTimestampMonotonic=0
 Transient=no
+StartLimitInterval=10000000
+StartLimitBurst=5
+StartLimitAction=none
```

**Comment by mischief** [comment]
was fixed by coreos/coreos-cloudinit#419
